src/test/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/JpaGoalPersistenceAdapterTest.kt

```diff
@@ -23,30 +25,80 @@ class JpaGoalPersistenceAdapterTest {
   private lateinit var persistenceAdapter: JpaGoalPersistenceAdapter
 
   @Mock
-  private lateinit var goalRepository: GoalRepository
+  private lateinit var actionPlanRepository: ActionPlanRepository
 
   @Mock
   private lateinit var goalMapper: GoalEntityMapper
 
   @Test
-  fun `should save goal`() {
+  fun `should create goal given action plan does not already exist`() {
     // Given
-    val domainGoal = aValidGoal()
+    val prisonNumber = aValidPrisonNumber()
+    val reference = UUID.randomUUID()
+    val domainGoal = aValidGoal(
+      reference = reference,
+    )
+
+    given(actionPlanRepository.findByPrisonNumber(any())).willReturn(null)
```

Comment:
Just something to note - I'm adding `JpaActionPlanPersistenceAdapter` on my branch and it made me wonder if this class should call the repository directly, or the adapter. I'm not sure if it really matters, but it's perhaps something to ponder.

Reply:
How do you mean? Which class?

Reply:
Discussed on Slack and decided to leave this as this is (at least for now).